src/main/java/org/folio/service/finance/transaction/TransactionService.java

```diff
@@ -138,9 +138,12 @@ public Future<Void> batchDelete(List<String> transactionIds, RequestContext requ
 
   public Future<Void> batchReleaseAndDelete(List<Transaction> transactions, RequestContext requestContext) {
     // NOTE: we will have to use transactionPatches when it is available (see MODORDERS-1008)
-    transactions.forEach(tr -> tr.getEncumbrance().setStatus(Encumbrance.Status.RELEASED));
-    List<String> ids = transactions.stream().map(Transaction::getId).toList();
-    return batchAllOrNothing(null, transactions, ids, null, requestContext);
+    List<String> allIds = transactions.stream().map(Transaction::getId).toList();
+    List<Transaction> transactionsToRelease = transactions.stream()
+      .filter(tr -> tr.getEncumbrance().getStatus() != Encumbrance.Status.RELEASED)
+      .toList();
+    transactionsToRelease.forEach(tr -> tr.getEncumbrance().setStatus(Encumbrance.Status.RELEASED));
+    return batchAllOrNothing(null, transactionsToRelease, allIds, null, requestContext);
```

Comment:
please add comment in code why we need to release and after these delete the same transactions.
I assume because we need to update budgets, but I think better to have a comment here

Reply:
Added a comment. We should probably add logic in mod-finance-storage to release encumbrances before they are deleted, but I don't think this is in the scope of MODFISTO-432 (which is already pretty big).